### PR TITLE
codereview fixes

### PR DESCRIPTION
**Pull request by glensc**
I looked over the codebase before deciding to use it.

made some fixes/improvements. comments included with each commit message.

### CODE REVIEW

**Comment by pitbulk** [comment]
thanks, I will review and merge them next week.

**Comment by pitbulk** [comment]
I merged your changes on master at my local.. and 2 test failed:

1) OneLogin_Saml_AuthRequestTest::testGetMetadataValidTimestamp
Failed asserting that 1500669286 matches expected 1500626086.

/var/www/newonelogin/tests/src/OneLogin/Saml/AuthRequestTest.php:74

2) OneLogin_Saml_MetadataTest::testGetMetadataValidTimestamp
Failed asserting that 1501274086 matches expected 1501230886.

/var/www/newonelogin/tests/src/OneLogin/Saml/MetadataTest.php:98

**Comment by simonberger** [comment]
There is a mistake in the date format translation.

```
strftime("%Y-%m-%dT%H:%M:%SZ", $time);
```

to


```
$date->format("Y-m-d\Th:i:s\Z");
```

in three files.

It has to be

```
$date->format("Y-m-d\TH:i:s\Z");
```

File: lib/Saml/AuthRequest.php [hunk 1]
```diff
@@ -56,10 +56,8 @@ protected function _generateUniqueID()
      */
     protected function _getTimestamp()
     {
-        $defaultTimezone = date_default_timezone_get();
-        date_default_timezone_set('UTC');
-        $timestamp = strftime("%Y-%m-%dT%H:%M:%SZ");
-        date_default_timezone_set($defaultTimezone);
+        $date = new DateTime('now', new DateTimeZone('UTC'));
+        $timestamp = $date->format("Y-m-d\Th:i:s\Z");
```
Comment (simonberger) [inline lib/Saml/AuthRequest.php]: Wrong hour format

File: lib/Saml/Metadata.php [hunk 2]
```diff
@@ -30,10 +30,9 @@ public function getXml()
      */
     protected function _getMetadataValidTimestamp()
     {
-        $timeZone = date_default_timezone_get();
-        date_default_timezone_set('UTC');
-        $time = strftime("%Y-%m-%dT%H:%M:%SZ", time() + self::VALIDITY_SECONDS);
-        date_default_timezone_set($timeZone);
+        $timestamp = time() + self::VALIDITY_SECONDS;
+        $date = new DateTime("@$timestamp", new DateTimeZone('UTC'));
+        $time = $date->format("Y-m-d\Th:i:s\Z");
```
Comment (simonberger) [inline lib/Saml/Metadata.php]: Wrong hour format

File: lib/Saml2/Utils.php [hunk 3]
```diff
@@ -658,10 +657,8 @@ public static function generateUniqueID()
      */
     public static function parseTime2SAML($time)
     {
-        $defaultTimezone = date_default_timezone_get();
-        date_default_timezone_set('UTC');
-        $timestamp = strftime("%Y-%m-%dT%H:%M:%SZ", $time);
-        date_default_timezone_set($defaultTimezone);
+        $date = new DateTime("@$time", new DateTimeZone('UTC'));
+        $timestamp = $date->format("Y-m-d\Th:i:s\Z");
```
Comment (simonberger) [inline lib/Saml2/Utils.php]: Wrong hour format

**Comment by pitbulk** [comment]
@glensc can you fix the time issue and rebase?

**Comment by LukasReschke** [comment]
Sorry for hijacking this but I've seen there's a 3.0.0 branch in the repository.

@pitbulk I'm planning on sending a Pull Request integrating https://github.com/etsy/phan as CI step here and also the appropriate fixes for this. Would you prefer this against the 3.0.0 branch or master? What's the policy on this repository? :)

**Comment by pitbulk** [comment]
@LukasReschke what advantages has phan vs Travis? What changes requires on the repo code?

**Comment by simonberger** [comment]
Can you please just create an issue for this discussion not in any relation to this pull request?

File: lib/Saml2/Utils.php (hunk 3 above)
Comment (glensc) [inline lib/Saml2/Utils.php]: fixed in e818a43, not sure why my tests passed when i pushed out this initially

**Comment by glensc** [comment]
rebased and force pushed

**Comment by pitbulk** [comment]
Thanks, it seems github-travis integration right now having issues.

I will review in a while and merge this improvement on 2.X and 3.0.0 branch.

**Comment by glensc** [comment]
yes. github is having technical issues. i had to post comments twice here in web ui too :)

https://status.github.com/ reports:
> 19:32 EEST Major service outage.

**Comment by simonberger** [comment]
You can restart the failed travis process.
https://travis-ci.org/onelogin/php-saml/builds/259441589

**Comment by pitbulk** [comment]
That build seems canceled. I clicked in order to restart it, but nothing happens.

**Comment by simonberger** [comment]
Too bad.
There seems no "restart failed process test" in the travis deployment routine... ;)

**Comment by glensc** [comment]
i as contributor do not have restart option in travis; however i amended last commit and force pushed to trigger new build. hope this fixes it.

**Comment by glensc** [comment]
yah. thanks!

**Pull request merged**